### ISSUE TITLE
Here, How to use convolution1D to deal series characters?

### ISSUE THREAD

**Issue by xyzhang16** [fchollet/keras#3077]
I have a five dimentions series vector. I want to use Convolution1D to get series characters
but I have some problems. I couldn't understand the difference about input_dim and input _length in Convoltion1D. 
coding:
`
# training set

x_test=np.random.random_sample((100,5))
y_test=np.random.randint(2,size=100)
# model

model= Sequential()
model.add(Convolution1D(nb_filter=1,filter_length=2,input_dim=5))
model.add(Flatten())
model.add(Dense(1))
model.compile(optimizer='sgd', loss='mse')
model.fit(x_test,y_test)
`
but it is wrong, It says I don't input parameter 'input_length'
So I change the code
code is following:
# training set

x_test=np.random.random_sample((100,5))
y_test=np.random.randint(2,size=100)
# model

model= Sequential()
model.add(Convolution1D(nb_filter=1,filter_length=2,input_dim=5,input_dim=5))
model.add(Flatten())
model.add(Dense(1))
model.compile(optimizer='sgd', loss='mse')
model.fit(x_test,y_test)
`
However, it is still wrong, It says my input-dimensions is wrong?

anyone know it?

**Comment by ChristianThomae** [comment]
You're nearly there, have a look at the documentation of the [Convolution1D layer](http://keras.io/layers/convolutional/#convolution1d), especially the part about the input shape.
It is expecting a 3-dim input: `(samples, steps, input_dim)`.
You're currently providing it a 2-dim input `(steps, input_dim)`: `x_test=np.random.random_sample((100,5))` **(100,5)**.

Just use something like:
```python
x_test=np.random.random_sample((1,100,5)) # 1 sample with 100 timesteps each of 5 features/dims
y_test=np.random.randint(2,size=1) # target for 1 sample

model = Sequential()
model.add(Convolution1D(nb_filter=1,filter_length=2,input_dim=5,input_length=100)) # 100 timesteps of 5 features each
model.add(Flatten())
model.add(Dense(1))
model.compile(optimizer='sgd', loss='mse')
model.fit(x_test,y_test)
```

**Comment by xyzhang16** [comment]
@ChristianThomae 
thank you for your help, I get it.
